tools/clang/unittests/HLSLExec/ShaderOpArith.xml

```diff
@@ -246,7 +246,7 @@
 #undef DISPATCHY
 #undef DISPATCHZ
 
-#define DISPATCHX 1
+#define DISPATCHX 4
```

Comment:
"dispatch" is the wrong term.  This isn't the dispatch dimensions but the threadgroup dimensions that we're talking about.

Reply:
Presumably, this block of code is just to make the MSMain compile when thread dimensions are too high?
Also, presumably we aren't using ASMain or MSMain under these conditions?

First, I don't know if ASMain has this same constraint that MSMain has.  Though, the spec doesn't appear to mention AS in the restriction described [here](https://microsoft.github.io/DirectX-Specs/d3d/MeshShader.html#numthreads).

Second, can't the test avoid compiling ASMain and/or MSMain when supplying invalid dimensions specified in the first place?

Reply:
Correct. This define overrides are used when the shader source code is compiled as a compute shader which allows larger thread group sizes than AS or MS. The ASMain and MSMain are not used in this case. This is the tax for having a single HLSL source code for the test. We could separate the code into two sections in the `ShaderOpArith.xml`, one for CS and one for MS & AS, but then some of the shared code would have to be duplicated in both sections.